File: meshlink-sample/android/app/src/main/kotlin/ch/trancee/meshlink/proof/android/ProofBenchmarkTransport.kt
package ch.trancee.meshlink.proof.android

import java.util.Locale

internal enum class ProofBenchmarkTransport {
    MeshLink,
    GattPrototype,
}

internal fun parseProofBenchmarkTransport(rawValue: String?): ProofBenchmarkTransport {
    return when (rawValue?.lowercase(Locale.US)) {
        "gatt", "gattprototype", "gatt-prototype" -> ProofBenchmarkTransport.GattPrototype
        else -> ProofBenchmarkTransport.MeshLink
    }
}

File: meshlink-sample/ios/ProofApp/ProofBenchmarkTransport.swift
import Foundation

enum ProofBenchmarkTransport {
    case meshLink
    case gattPrototype

    static func parse(_ rawValue: String?) -> ProofBenchmarkTransport {
        switch rawValue?.lowercased() {
        case "gatt", "gattprototype", "gatt-prototype":
            return .gattPrototype
        default:
            return .meshLink
        }
    }

    var logLabel: String {
        switch self {
        case .meshLink:
            return "meshlink"
        case .gattPrototype:
            return "gattPrototype"
        }
    }
}
